### PR TITLE
[gitlab] Refactor package build scripts

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

First pass at simplifying a few bits of the package build scripts in the Gitlab CI:
- moves the RPM GPG key setup step to a common script,
- moves the system-probe artifacts fetching and extraction to a common script,
- removes outdated comments.

### Motivation

Avoid potential drift in multiple places that do the same thing, and make reading the CI scripts easier.

### Additional Notes

n/a

### Possible Drawbacks / Trade-offs

n/a

### Describe how to test/QA your changes

Verify that all package builds are still working.

### CODE REVIEW

**Comment by KSerrania** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: .gitlab/package_build/deb.yml [hunk 1]
```diff
@@ -148,10 +144,7 @@ agent_deb-arm64-a7:
     - echo "Target architecture ${DD_TARGET_ARCH:=amd64}"
     # remove artifacts from previous pipelines that may come from the cache
     - rm -rf $OMNIBUS_PACKAGE_DIR/*
-    # Artifacts and cache must live within project directory but we run omnibus in a neutral directory.
-    # Thus, we move the artifacts at the end in a gitlab-friendly dir.
     - !reference [.setup_deb_signing_key]
-    # Use --skip-deps since the deps are installed by `before_script`.
```
Comment (alopezz) [inline .gitlab/package_build/deb.yml:154]: If this is no longer true, do we know why we still use `--skip-deps`?

**Review by alopezz — APPROVED** [review]
Thanks for this!

File: .gitlab/package_build/deb.yml (hunk 1 above)
Comment (KSerrania) [reply .gitlab/package_build/deb.yml:154]: This one is still true, but:
- I think it's easy to read why it is used (the `- !reference [.retrieve_linux_go_deps]` part is a few lines above)
- I find it weird that there is only a comment for that specific option, and not all the other ones that fall in the same pattern (eg. `--go-mod-cache`).

So I proposed removing it (assuming that reading the job script is enough to understand), but another option is to put it back, and add the explanations for the other options as well, to be consistent.

Comment (alopezz) [reply .gitlab/package_build/deb.yml:154]: 👍 Removing it is fine.

File: .gitlab/common/system_probe.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+
```
Comment (CelianR) [inline .gitlab/common/system_probe.yml:2]: **🥜 nitpick:** ‏
```suggestion
```

**Review by CelianR — APPROVED** [review]

**Pull request closed without merging**